external_search.py

```diff
@@ -1983,44 +1992,140 @@ def sort_order(self):
+                # The simple case, handled below.
+                pass
+
+        if '.' not in key:
+            # A simple case.
+            return { key : self.asc }
+
+        # At this point we're sorting by a nested field.
+        nested = None
+        if key == 'licensepools.availability_time':
+            # We're sorting works by the time they became
+            # available to a library. This means we only want to
+            # consider the availability times of license pools
+            # found in one of the library's collections.
+            collection_ids = self._filter_ids(self.collection_ids)
+            if collection_ids:
+                nested = dict(
+                    path="licensepools",
+                    filter=dict(
+                        terms={
+                            "licensepools.collection_id": collection_ids
+                        }
+                    ),
                 )
-            sort_description = dict(order=ascending, mode=mode)
-            if nested:
-                sort_description['nested'] = nested
-            order = { key : sort_description }
+
+            # If a book shows up in multiple collections, we're only
+            # interested in the collection that had it the earliest.
+            mode = 'min'
         else:
-            order = { key : ascending }
+            raise ValueError(
+                "I don't know how to sort by %s." % key
+            )
+        sort_description = dict(order=self.asc, mode=mode)
+        if nested:
+            sort_description['nested']=nested
+        return { key : sort_description }
+
+    @property
+    def _last_update_time_order_by(self):
+        """We're sorting works by the time of their 'last update'.  An
+        'update' happens when the work's metadata is changed, when
+        it's added to a collection used in this Filter, or when it's
+        added to one of the lists used in this Filter.
+
+        This is complex enough that we need to write a script that runs
+        on the Elasticsearch side to calculate the last update for
+        any given work.
+        """
+
+        # First, set up the parameters we're going to pass into the
+        # script -- a list of custom list IDs relevant to this filter,
+        # and a list of collection IDs relevant to this filter.
+        collection_ids = self._filter_ids(self.collection_ids)
+
+        # The different restriction sets don't matter here. The filter
+        # part of the query ensures that we only match works present
+        # on one list in every restriction set. Here, we need to find
+        # the latest time a work was added to _any_ relevant list.
+        all_list_ids = set()
+        for restriction in self.customlist_restriction_sets:
+            all_list_ids.update(self._filter_ids(restriction))
+        nested = dict(
+            path="customlists",
+            filter=dict(
+                terms={"customlists.list_id": sorted(all_list_ids)}
```

Comment:
does it need to be sorted?

Reply:
No, I was thinking this would make it easier to test but I don't actually need it in the test.